### PR TITLE
New package: libresprite-1.0

### PR DESCRIPTION
**Pull request by moheb2000**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)

### CODE REVIEW

File: srcpkgs/libresprite/template [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Template file for 'libresprite'
+pkgname=libresprite
+version=1.0
+revision=1
+archs="x86_64"
```
Comment (Chocimier) [inline srcpkgs/libresprite/template:5]: Nope. Unmark, rename hostmakedepends to makedepends and let's then look how to fix musl, i686 and arm.

File: srcpkgs/libresprite/template [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE:STRING=Release -DWITH_DESKTOP_INTEGRATION=ON"
+hostmakedepends="git zlib-devel libpng-devel freetype-devel giflib-devel gtest-devel libjpeg-turbo-devel libcurl-devel SDL2-devel SDL2_image-devel lua-devel pixman-devel tinyxml-devel nodejs-devel"
+short_desc="Animated sprite editor & pixel art tool"
+maintainer="Mohammad Ebrahimi <moh.ebrahimi2000@gmail.com>"
+license="GPL-2.0-or-later"
+homepage="https://libresprite.github.io"
+
+do_fetch() {
+	git clone --recursive https://github.com/LibreSprite/LibreSprite.git $wrksrc
```
Comment (Chocimier) [inline srcpkgs/libresprite/template:16]: We want fast rebuild without cloning every time. List submodules like in ppsspp does or ask upstream for combined tarball.

File: srcpkgs/libresprite/template [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Template file for 'libresprite'
+pkgname=libresprite
+version=1.0
+revision=1
+archs="x86_64"
+create_wrksrc=yes
+build_style=cmake
+configure_args="-DCMAKE_BUILD_TYPE:STRING=Release -DWITH_DESKTOP_INTEGRATION=ON"
```
Comment (Chocimier) [inline srcpkgs/libresprite/template:8]: CMAKE_BUILD_TYPE:STRING=Release disables debug symbols, anything wrong with void's default?

File: srcpkgs/libresprite/template [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	git reset --hard v1.0
+}
+
+post_install() {
+	vdoc ${wrksrc}/docs/files/ase.txt
+	vdoc ${wrksrc}/docs/files/fli.txt
+	vdoc ${wrksrc}/docs/files/msk.txt
+	vdoc ${wrksrc}/docs/files/pic.txt
+	vdoc ${wrksrc}/docs/files/picpro.txt
+	vlicense ${wrksrc}/LICENSE.txt
```
Comment (Chocimier) [inline srcpkgs/libresprite/template:27]: As xlint point out, no need to include GPL.

**Comment by moheb2000** [comment]
Thanks!

I fixed the problems but there are some other problems that I want to say here:

- clip test units will fail but clipboard functionality works just fine like the official libresprite AppImage.
- I tried to cross compile this package for armv6l but nodejs-devel package isn't in void official repositories. it's not required based on what I understood from cmake config but I still can't cross compile it because of this error:
```
[307/723] Generating brush_slot_params.xml.h
FAILED: src/app/brush_slot_params.xml.h /builddir/libresprite-1.0/build/src/app/brush_slot_params.xml.h 
cd /builddir/libresprite-1.0/build/src/app && /builddir/libresprite-1.0/build/bin/gen --input /builddir/libresprite-1.0/data/widgets/brush_slot_params.xml --widgetid brush_slot_params > /builddir/libresprite-1.0/build/src/app/brush_slot_params.xml.h.tmp && /usr/bin/cmake -E copy_if_different /builddir/libresprite-1.0/build/src/app/brush_slot_params.xml.h.tmp /builddir/libresprite-1.0/build/src/app/brush_slot_params.xml.h
/bin/sh: line 1: /builddir/libresprite-1.0/build/bin/gen: cannot execute binary file: Exec format error
[308/723] Generating canvas_size.xml.h
```
So if compiling for arm is necessary, I need to figure out what's the problem with this error.

**Comment by moheb2000** [comment]
The clip tests depended to `xvfb-run` package and I fixed them. I will fix cross compilation problem and then I will commit it. Thanks a lot.

**Comment by moheb2000** [comment]
It's still **not** ready for build!
I changed some makefiles and build generator code in post_extract function but still I get that error. native compiling has no problem!

**Pull request closed without merging**